app/components/time-picker.js

```diff
@@ -2,11 +2,24 @@ import moment from 'moment';
 import Ember from 'ember';
 import momentFormat from 'ember-moment/computeds/format';
 
+const { computed } = Ember;
+const { oneWay } = computed;
+
 export default Ember.Component.extend({
   classNames: ['time-picker'],
   date: null,
-  dateBuffer: Ember.computed.oneWay('date'),
-  hour: momentFormat('dateBuffer', 'h'),
+  dateBuffer: oneWay('date'),
+
+  hour: computed('dateBuffer', function() {
+    const defaultHour = this.get('defaultHour');
```

Comment:
I think the default hour stuff shouldn't be part of this component.  When it is created it gets passed a data, if the calling component wants a default then it should make one up before calling time-picker.